patches/node/fix_key_gen_apis_are_not_available_in_boringssl.patch

```diff
@@ -5,72 +5,20 @@ Subject: fix: key gen APIs are not available in BoringSSL
-     V(RAND)                                                                   \
--    V(DSO)                                                                    \
-     V(ENGINE)                                                                 \
-     V(OCSP)                                                                   \
-     V(UI)                                                                     \
-     V(COMP)                                                                   \
-     V(ECDSA)                                                                  \
-     V(ECDH)                                                                   \
--    V(OSSL_STORE)                                                             \
--    V(FIPS)                                                                   \
--    V(CMS)                                                                    \
--    V(TS)                                                                     \
-     V(HMAC)                                                                   \
--    V(CT)                                                                     \
--    V(ASYNC)                                                                  \
--    V(KDF)                                                                    \
--    V(SM2)                                                                    \
-     V(USER)                                                                   \
- 
- #define V(name) case ERR_LIB_##name: lib = #name "_"; break;
-@@ -6138,6 +6128,7 @@ class DSAKeyPairGenerationConfig : public KeyPairGenerationConfig {
-     if (EVP_PKEY_paramgen_init(param_ctx.get()) <= 0)
-       return nullptr;
- 
-+#ifndef OPENSSL_IS_BORINGSSL
-     if (EVP_PKEY_CTX_set_dsa_paramgen_bits(param_ctx.get(), modulus_bits_) <= 0)
-       return nullptr;
- 
-@@ -6148,6 +6139,7 @@ class DSAKeyPairGenerationConfig : public KeyPairGenerationConfig {
-         return nullptr;
-       }
-     }
-+#endif
- 
-     EVP_PKEY* raw_params = nullptr;
-     if (EVP_PKEY_paramgen(param_ctx.get(), &raw_params) <= 0)
-diff --git a/src/node_crypto_common.cc b/src/node_crypto_common.cc
-index 6473b652ac95609aff555d99be38b48a5aa513a5..caaaf19dc02101c2024b511780c94fc85476b7a2 100644
---- a/src/node_crypto_common.cc
-+++ b/src/node_crypto_common.cc
-@@ -240,10 +240,10 @@ int UseSNIContext(const SSLPointer& ssl, BaseObjectPtr<SecureContext> context) {
- }
- 
- const char* GetClientHelloALPN(const SSLPointer& ssl) {
-+#ifndef OPENSSL_IS_BORINGSSL
+diff --git a/src/crypto/crypto_common.cc b/src/crypto/crypto_common.cc
+index f4b7bd3ad8548a0b69943ddea669e6f1991b7a49..221d652fa7de246e5f69fcf392e334087bac0199 100644
+--- a/src/crypto/crypto_common.cc
++++ b/src/crypto/crypto_common.cc
+@@ -242,7 +242,7 @@ const char* GetClientHelloALPN(const SSLPointer& ssl) {
    const unsigned char* buf;
    size_t len;
    size_t rem;
 -
++#ifndef OPENSSL_IS_BORINGSSL
```

Comment:
How come this `#ifndef` is moved to after the variables? Doesn't that make the variables warn as unused...?